constrained_linear_regression/__init__.py

```diff
@@ -100,7 +100,7 @@ def fit(self, X, y, initial_beta=None):
             accept_sparse=['csr', 'csc', 'coo'],
             y_numeric=True, multi_output=False
         )
-        X, y, X_offset, y_offset, X_scale = self._preprocess_data(
```

Comment:
With which version of scikit-learn did you test this?
The method `self._preprocess_data` used to work for me when I implemented this. 
If `scikit-learn` API has changed at some moment, we probably need to find the version where it happened and reflect it in the requirements of `constrained-linear-regression`.

Reply:
I am currently using `scikit-learn==1.2.2`, and it works fine.

Reply:
First, when I upgrade sciket-learn>1.1, this error happens.
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
constrained-linear-regression 0.0.4 requires scikit-learn<=1.1, but you have scikit-learn 1.2.2 which is incompatible.
```
```bash
pip install -U scikit-learn==1.2.2
pip list | grep -E "(constrained|scikit)"
```
```
constrained-linear-regression 0.0.4
scikit-learn                  1.2.2
```
Run the reproducing error code at #4 
```error
X, y, X_offset, y_offset, X_scale = self._preprocess_data(
AttributeError: 'MatrixConstrainedLinearRegression' object has no attribute '_preprocess_data'
```